DevelopingSpace/starchart
# Add new DNS Record form tests

Resolves #278, resolves #296 

Since writing the tests relied on setting up the shared auth session, these have been combined into one PR. The following changes have been made:
- Added auth setup file for e2e tests to login as a user and save session data in a JSON file
- Updated playwright config to add a setup project which uses the auth setup file. Added this setup project as a dependency to all other projects. This means the setup will run before the tests are run
- Added above JSON file to `.gitignore`. This file is generated before the tests are run so it is not necessary to save it
- Added tests for new DNS Record form. Here the saved session generated by the setup is used for the authenticated tests
- Updated Mobile Safari config as the tests pass locally but fail on CI, as per comment below

### Steps to test
You can check the CI to confirm that the tests pass. Alternatively,
- Ensure that app is setup according to [setup instructions](https://github.com/DevelopingSpace/starchart/blob/main/CONTRIBUTING.md#development)
- Run `npm run docker`
- Run `npm run build`
- Run `npm run test:e2e:run`
- Check that tests added in this PR pass. **Note**: Some of the tests for Login will fail locally. This is not related to this PR, and can be investigated in another issue
